govcd/openapi_org_network.go

```diff
@@ -0,0 +1,270 @@
+func (orgVdcNet *OpenApiOrgVdcNetwork) IsRouted() bool {
+	return orgVdcNet.GetType() == types.OrgVdcNetworkTypeRouted
+}
+
+// IsImported returns true if the network type is imported (NSX-T only)
+func (orgVdcNet *OpenApiOrgVdcNetwork) IsImported() bool {
+	return orgVdcNet.GetType() == types.OrgVdcNetworkTypeOpaque
+}
+
+// IsDirect returns true if the network type is direct (NSX-V only)
+func (orgVdcNet *OpenApiOrgVdcNetwork) IsDirect() bool {
+	return orgVdcNet.GetType() == types.OrgVdcNetworkTypeDirect
+}
+
+// getOpenApiOrgVdcNetworkById is a private parent for wrapped functions:
+// func (org *Org) GetOpenApiOrgVdcNetworkById(id string) (*OpenApiOrgVdcNetwork, error)
+// func (vdc *Vdc) GetOpenApiOrgVdcNetworkById(id string) (*OpenApiOrgVdcNetwork, error)
+func getOpenApiOrgVdcNetworkById(client *Client, id string, queryParameters url.Values) (*OpenApiOrgVdcNetwork, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointOrgVdcNetworks
+	minimumApiVersion, err := client.checkOpenApiEndpointCompatibility(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if id == "" {
+		return nil, fmt.Errorf("empty Org VDC network id")
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint, id)
+	if err != nil {
+		return nil, err
+	}
+
+	egw := &OpenApiOrgVdcNetwork{
+		OpenApiOrgVdcNetwork: &types.OpenApiOrgVdcNetwork{},
+		client:               client,
+	}
+
+	err = client.OpenApiGetItem(minimumApiVersion, urlRef, queryParameters, egw.OpenApiOrgVdcNetwork)
+	if err != nil {
+		return nil, err
+	}
+
+	return egw, nil
+}
+
+// returnSingleOpenApiOrgVdcNetwork helps to reduce code duplication for `GetOpenApiOrgVdcNetworkByName` functions with different
+// receivers
+func returnSingleOpenApiOrgVdcNetwork(name string, allEdges []*OpenApiOrgVdcNetwork) (*OpenApiOrgVdcNetwork, error) {
+	if len(allEdges) > 1 {
+		return nil, fmt.Errorf("got more than 1 Org Vdc network by name '%s' %d", name, len(allEdges))
+	}
+
+	if len(allEdges) < 1 {
+		return nil, fmt.Errorf("%s: got 0 Org Vdc network by name '%s'", ErrorEntityNotFound, name)
+	}
+
+	return allEdges[0], nil
+}
+
+// getAllOpenApiOrgVdcNetworks is a private parent for wrapped functions:
+// func (vdc *Vdc) GetAllOpenApiOrgVdcNetworks(queryParameters url.Values) ([]*OpenApiOrgVdcNetwork, error)
+//
+// Note. If pageSize > 32 it will be limited to maximum of 32 in this function because API validation does not allow
+// higher number
+func getAllOpenApiOrgVdcNetworks(client *Client, queryParameters url.Values) ([]*OpenApiOrgVdcNetwork, error) {
+
+	// Enforce maximum pageSize to be 32 as API endpoint throws error if it is > 32
+	pageSizeString := queryParameters.Get("pageSize")
+
+	switch pageSizeString {
+	// If no pageSize is specified it must be set to 32 as by default low level API function OpenApiGetAllItems sets 128
+	case "":
+		queryParameters.Set("pageSize", "32")
+
+		// If pageSize is specified ensure it is not >32
```

Comment:
This comment seems misplaced.

Reply:
This is in correct place. I might need to improve the comment then. The point is that above `case ""` sets pageSize to 32 if none was specified.
The `default` case will check what value was provided and if it is `>32` it will set it to `32`. This is needed because this particular endpoint throws an ugly error if `pageSize>32`

Reply:
Ah, so then the indentation is too big, as it should be inline with `default:` as right now it looks as if the comment belongs to the `case "":`